ovs/lib/helpers/decorators.py

```diff
@@ -172,8 +197,13 @@ def update_value(key, append, value_to_update=None):
                     for task in task_names:
                         key_to_check = '{0}_{1}'.format(ENSURE_SINGLE_KEY, task)
                         if persistent_client.exists(key_to_check):
-                            log_message('Execution of task {0} discarded'.format(task_name))
-                            return None
+                            if callback is None:
+                                log_message('Execution of task {0} discarded'.format(task_name))
+                                return None
+                            else:
+                                log_message('Execution of task {0} in progress, executing callback function {1}'.format(task_name, callback.__name__))
+                                return callback(*args, **kwargs)
```

Comment:
Does this actually works if you're running the task async on a different node? I'd think it can't reach the callback function?